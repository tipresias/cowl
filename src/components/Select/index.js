// @flow
import React from 'react';
import type { Node } from 'react';

type Props = {
  name: string,
  value: number,
  onChange: (event: SyntheticEvent<HTMLSelectElement>) => void,
  options: Array<number>
}
const Select = ({
  name,
  value,
  onChange,
  options,
}: Props): Node => (
  <select
    name={name}
    value={value}
    onChange={onChange}
  >
    {
      options.map(option => (
        <option key={option} value={option}>
          {option}
        </option>
      ))
    }
  </select>
);

export default Select;
