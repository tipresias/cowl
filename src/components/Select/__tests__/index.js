import React from 'react';
import { shallow } from 'enzyme';
import Select from '../index';

describe('select', () => {
  it.skip('renders', () => {
    // arrange
    const optionsMocked = ['item1', 'item2', 'item3'];
    const valueMocked = 'mocked_value';
    const onChangeSpy = jest.fn();

    // act
    const wrapper = shallow(<Select
      value={valueMocked}
      onChange={onChangeSpy}
      options={optionsMocked}
    />);

    // assert
    expect(wrapper).toMatchSnapshot();
  });

  it('calls onChange callback when select onChange is executed', () => {
    const optionsMocked = ['item1', 'item2', 'item3'];
    const valueMocked = 'mocked_value';
    const onChangeSpy = jest.fn();
    const eventObjectMocked = { target: { value: 'ok' } };
    const wrapper = shallow(<Select
      value={valueMocked}
      onChange={onChangeSpy}
      options={optionsMocked}
      name="test_select"
      id="test_select1"
      label="choose an option"
    />);
    wrapper.find('Select__SelectStyled').simulate('change', eventObjectMocked);
    expect(onChangeSpy).toHaveBeenCalledWith({ target: { value: 'ok' } });
  });

  it('sets the value prop when value prop is passed', () => {
    const optionsMocked = ['item1', 'item2', 'item3'];
    const valueMocked = 'mocked_value';
    const onChangeSpy = jest.fn();
    const wrapper = shallow(<Select
      value={valueMocked}
      onChange={onChangeSpy}
      options={optionsMocked}
      name="test_select"
      id="test_select1"
      label="choose an option"
    />);
    expect(wrapper.find('Select__SelectStyled').props().value).toBe('mocked_value');
  });
});
