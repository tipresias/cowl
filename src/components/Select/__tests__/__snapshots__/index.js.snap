// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['select renders 1'] = 'ShallowWrapper {}';
